src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/loptimizer/heuristics/utils/Join_Analyzer.java

```diff
@@ -0,0 +1,53 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.loptimizer.heuristics.utils;
+
+import org.apache.jena.graph.Node;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Join_Analyzer {
+
+    public static double getStarJoins( final List<Node> vars_s, final List<Node> vars_o, final double J_Ts ){
+        final int jsubs = countDuplicates(vars_s);
+        final int jobjs = countDuplicates(vars_o);
+        return ( jsubs + jobjs ) * J_Ts;
+    }
+
+    public static double getChainJoins( final List<Node> vars_s, final List<Node> vars_o, final double J_Tc ) {
+        final int chainjoins = countDuplicates(vars_s,vars_o);
+        return chainjoins * J_Tc;
+    }
+
+    public static double getUnusualJoins( final List<Node> vars_s, final List<Node> vars_p, final List<Node> vars_o, final double J_Tu ) {
```

Comment:
These three function names are misleading because the functions do not actually return any joins.